tests/common/reboot.py

```diff
@@ -495,3 +495,45 @@ def check_reboot_cause_history(dut, reboot_type_history_queue):
     logger.error("The number of expected reboot-cause:{} is more than that of actual reboot-cuase:{}".format(
         reboot_type_history_len, len(reboot_type_history_queue)))
     return False
+
+
+def check_determine_reboot_cause_service(dut):
+    """
+    @summary: This function verifies the status of the 'determine-reboot-cause' service on the device under test (DUT).
+    It checks:
+    1. The service's ActiveState and SubState using systemctl.
+    2. The kernel logs (dmesg) for critical errors that may indicate service failures or system instability.
+    @param dut: The AnsibleHost object of DUT.
+    """
+    # Step 1: Check the 'determine-reboot-cause' service status
+    logger.info("Checking 'determine-reboot-cause' service status using systemctl")
+    service_state = dut.get_service_props("determine-reboot-cause.service")
+
+    # Validate service is active
+    active_state = service_state.get("ActiveState", "")
+    sub_state = service_state.get("SubState", "")
+    logger.info(f"'determine-reboot-cause' ActiveState: {active_state}, SubState: {sub_state}")
+
+    assert active_state == "active", f"Service 'determine-reboot-cause' is not active. Current state: {active_state}"
+    assert sub_state == "exited", f"Service 'determine-reboot-cause' did not exit cleanly. \
+            Current sub-state: {sub_state}"
+
+    # Step 2: Check kernel logs (dmesg) for any critical errors
+    logger.info("Checking dmesg for critical errors")
+    dmesg_output = dut.command("sudo dmesg")["stdout"]
+
+    # List of error keywords to look for in the dmesg logs
+    error_keywords = ["crash", "Out of memory", "Call Trace", "Exception", "panic", "segfault"]
```

Comment:
Wouldn't this part be better served by loganalyzer?

Reply:
@saiarcot895 Since these are reboot tests, the logs will rotate upon reboot. Adding loganalyzer would require significant effort. If you believe it adds value, I can update the code to incorporate loganalyzer changes.

Reply:
@saiarcot895 for now I have removed this additional step to check for errors which is not in the scope of PR and will be handled separately. Thank you!